pandas/_libs/algos.pyx

```diff
@@ -249,44 +252,82 @@ def nancorr(ndarray[float64_t, ndim=2] mat, bint cov=0, minp=None):
     result = np.empty((K, K), dtype=np.float64)
     mask = np.isfinite(mat).view(np.uint8)
 
-    with nogil:
-        for xi in range(K):
-            for yi in range(xi + 1):
-                nobs = sumxx = sumyy = sumx = sumy = 0
-                for i in range(N):
-                    if mask[i, xi] and mask[i, yi]:
-                        vx = mat[i, xi]
-                        vy = mat[i, yi]
-                        nobs += 1
-                        sumx += vx
-                        sumy += vy
+    if parallel:
+        with nogil:
+            for xi in prange(K, schedule='dynamic'):
+                nancorr_single_row(mat, N, K, result, xi, mask, minpv, cov)
+    else:
+        with nogil:
+            for xi in range(K):
```

Comment:
could this be collapsed with something like `range_func = range(K) if not parallel else prange(K, schedule='dynamic')`?

Reply:
AFAIK No, because prange is not really a function, it is converted into a #pragma of sorts in the c code.

Reply:
OK.  I think there is an issue on Cython's GH about making a maybe_nogil that would be semantically similar to maybe_prange.  Might be worth commenting there